playground/frontend/playground_components/tools/extract_symbols_java/src/main/java/com/playground/extract_symbols/Main.java

```diff
@@ -0,0 +1,93 @@
+package com.playground.extract_symbols;
+
+import com.github.javaparser.ParseProblemException;
+import com.github.javaparser.StaticJavaParser;
+import com.github.javaparser.ast.body.ClassOrInterfaceDeclaration;
+import com.github.javaparser.ast.body.FieldDeclaration;
+import com.github.javaparser.ast.body.MethodDeclaration;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Main {
+    public static void main(String[] args) throws IOException {
+        var sdkPath = args[0];
+        HashMap<String, ClassInfo> classInfoMap = collectClassInfo(sdkPath);
+        String yamlString = buildYamlString(classInfoMap);
+        System.out.println(yamlString);
+    }
+
+    private static HashMap<String, ClassInfo> collectClassInfo(String sdkPath) throws IOException {
+        var classInfoList = new HashMap<String, ClassInfo>();
+        var paths = new File(sdkPath).toPath().toAbsolutePath();
+        Files.walk(paths).forEach(path -> {
+            var stringPath = path.toString();
+            if (stringPath.endsWith(".java") && !stringPath.contains("test")) {
```

Comment:
Please extract to something like
https://github.com/apache/beam/blob/58b4d46655d94374f3d3564752dc12eb98b95456/playground/frontend/playground_components/tools/extract_symbols_go/extract_symbols_go.go#L83

Reply:
A check for containing 'test' is not the grea**test** one.

Reply:
Agreed, but there are a lot of directories in java sdk. Looks like it's better to check it here than pass all directories we need